pkg/traces/remotewriteexporter/exporter.go

```diff
@@ -50,12 +50,14 @@ type remoteWriteExporter struct {
 func newRemoteWriteExporter(cfg *Config) (component.MetricsExporter, error) {
 	logger := log.With(util.Logger, "component", "traces remote write exporter")
 
-	ls := make(labels.Labels, 0, len(cfg.ConstLabels))
-	for _, constLabel := range cfg.ConstLabels {
-		ls = append(ls, labels.Label{
-			Name:  constLabel.Name,
-			Value: constLabel.Value,
-		})
+	var ls labels.Labels
+
+	if cfg.ConstLabels != nil {
+		ls = make(labels.Labels, 0, len(*cfg.ConstLabels))
+
+		for name := range *cfg.ConstLabels {
+			ls = append(ls, labels.Label{Name: name, Value: (*cfg.ConstLabels)[name]})
+		}
```

Comment:
This could be simplified to loop with the value as well

```suggestion
		for name, value := range *cfg.ConstLabels {
			ls = append(ls, labels.Label{Name: name, Value: value})
		}
```